### PR TITLE
Basic Pagination for Audio Entries

### PR DESCRIPTION
**Pull request by georgewwww**
Added paginated style for displaying audio entries.
Base class for paginated view is **PaginatedList**

### CODE REVIEW

File: NPlaylist/NPlaylist/appsettings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "ConnectionStrings": {
-    "ApplicationDbConnection": "Server=.\\SQLEXPRESS;Database=NPlaylist;Trusted_Connection=True;MultipleActiveResultSets=true",
-    "AuthenticationDbConnection": "Server=.\\SQLEXPRESS;Database=NPlaylist.Auth;Trusted_Connection=True;MultipleActiveResultSets=true"
+    "ApplicationDbConnection": "Server=EN32153;Database=NPlaylist_Application;Trusted_Connection=True;",
```
Comment (maksimenko-stanislav) [inline NPlaylist/NPlaylist/appsettings.json]: Please revert to `Server` to default.

File: NPlaylist/NPlaylist.Persistence/AudioEntries/SqlAudioEntriesRepository.cs [hunk 2]
```diff
@@ -1,29 +1,48 @@
...
+    public class SqlAudioEntriesRepository : SqlCrudRepository<Audio, Guid>, IAudioEntriesRepository
+    {
+        public SqlAudioEntriesRepository(ApplicationDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public async Task<bool> EntryExistsAsync(Guid audioId, CancellationToken ct)
+        {
+            var entity = await GetByIdAsync(audioId, ct);
+            return entity != null;
```
Comment (aivascu) [inline NPlaylist/NPlaylist.Persistence/AudioEntries/SqlAudioEntriesRepository.cs]: I don't think receiving `null` is an option with this method. A empty collection however is more than probable.

File: NPlaylist/NPlaylist/Models/PaginatedList.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (aivascu) [inline NPlaylist/NPlaylist/Models/PaginatedList.cs]: Cleanup the usings

File: NPlaylist/Tests/NPlaylist.Tests/Controllers/AudioControllerTests.cs [hunk 4]
```diff
@@ -10,16 +11,29 @@ namespace NPlaylist.Tests.Controllers
...
+        public void Index_ExpectedException_ReturnsOutOfRangeException()
+        {
+            var service = Substitute.For<IAudioService>();
+
+            var controller = new AudioControllerBuilder().WithAudioService(service)
+                .Build();
+
+            Func<Task> result = async ()
+                => await controller.Index(CancellationToken.None, 0);
+            result.Should().Throw<ArgumentOutOfRangeException>();
```
Comment (aivascu) [inline NPlaylist/Tests/NPlaylist.Tests/Controllers/AudioControllerTests.cs]: Add an empty line

File: NPlaylist/Tests/NPlaylist.Tests/Controllers/AudioControllerTests.cs [hunk 5]
```diff
@@ -10,16 +11,29 @@ namespace NPlaylist.Tests.Controllers
...
+
             var result = await controller.Index(CancellationToken.None) as ViewResult;
             result.Should().NotBeNull();
         }
+
+        [Fact]
+        public void Index_ExpectedException_ReturnsOutOfRangeException()
+        {
+            var service = Substitute.For<IAudioService>();
+
```
Comment (aivascu) [inline NPlaylist/Tests/NPlaylist.Tests/Controllers/AudioControllerTests.cs]: Remove the empty line. The three AAA parts should be visibly separated between eachother

File: NPlaylist/NPlaylist/Services/AudioService/AudioServiceImpl.cs [hunk 6]
```diff
@@ -23,10 +23,19 @@ public AudioServiceImpl(IMapper mapper, IAudioEntriesRepository audioEntriesRepo
...
+        public async Task<PaginatedList<AudioEntryViewModel>> GetAudioEntriesRangeAsync(int page, int count, CancellationToken ct)
         {
-            var entriesRepo = await _audioEntriesRepository.GetAllAsync(ct);
-            return _mapper.Map<List<AudioEntryViewModel>>(entriesRepo);
+            if (page < 1 || count < 1)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            var totalEntries = await _audioEntriesRepository.CountAsync(ct);
```
Comment (aivascu) [inline NPlaylist/NPlaylist/Services/AudioService/AudioServiceImpl.cs]: give the variables a more meaningful name `totalEntriesCount`, `pageEntries`, etc.

**Pull request merged**